compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -666,15 +666,17 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
                 isSubType(info1, info2)
 
             if defn.isFunctionType(tp2) then
-              tp1w.widenDealias match
-                case tp1: RefinedType =>
-                  return isSubInfo(tp1.refinedInfo, tp2.refinedInfo)
-                case _ =>
-            else if tp2.parent.typeSymbol == defn.PolyFunctionClass then
-              tp1.member(nme.apply).info match
-                case info1: PolyType =>
-                  return isSubInfo(info1, tp2.refinedInfo)
-                case _ =>
+              if defn.isPolyFunctionType(tp2) then
+                // TODO should we handle ErasedFunction is this same way?
+                tp1.member(nme.apply).info match
+                  case info1: PolyType =>
+                    return isSubInfo(info1, tp2.refinedInfo)
+                  case _ =>
+              else
```

Comment:
I am not sure why there is a difference between the two comparisons. Would it work to always use the second one, which tests for `RefinedType`?

Reply:
I will try to merge the two cases into one.